parquet-hadoop/src/main/java/org/apache/parquet/hadoop/CodecFactory.java

```diff
@@ -192,4 +105,23 @@ public void release() {
     }
     decompressors.clear();
   }
+
+  // May be able to allow this to stay abstract, check back with semver later
```

Comment:
this class is not meant to be extended outside Parquet.
The set of compression algorithms is fixed by the spec.
We should ignore that class in semver (even move it to its own package?) and have the base class we want:
- remove the constructor
- getCodecName should be implemented as there is always a valid one.
- compress/decompress should be abstract.